### PR TITLE
BUG: set Vmax to Voc at STC so extrapolation is always in 4th quad.

### LINKED ISSUE #36: make sure Vmax for all strings is greater than Vmp of system
**observed**: if some strings have lower Vmax than others, then linear extrapolation using the last 2 points in the string with the lower Vmax is used which is arbitrary. This is especially an issue if the practical operating point of the system (ie: Vmp) is at a voltage greater than Vmax of any of the strings. Then the current of the low Vmax strings is arbitrarily determined by the linear extrapolation and the uncertainty is unknown.

**proposed**: Set Vmax for all of the strings to be Voc at STC.

### PR DESCRIPTION
**Pull request by mikofski**
Fixes #36 so that extrapolation into 4th quadrant is more accurate by extending range of voltages all the way to Voc at STC where IV curve actually starts to turn downward, instead of Voc at `(Tc, Ee)` which might have the wrong slope.

* add new VocSTC attribute to PVcell
* set Aph to nan if Isc is zero
* add new _VocSTC method to calculate VocSTC attr
* add new Igen property and use everywhere instead of Isc * Aph
* use VocSTC as Vmax and make Vdiode from 3 sections includeing a new
Vquad4 section that spans from Voc to VocSTC with logspace concentrated
around Voc, unless Voc==VocSTC, in which case set Vquad4 from 0.8Voc to Voc
* update calc_series_test_iv.dat and other test values in test_pvcell and
test_setsuns
* set EPS as constant once
* use newton instead of fsolve for calcVcell and calcIcell
* add imagemagick converter to change travisci .svg to show up in pdf docs

Signed-off-by: Mark Mikofski <bwana.marko@yahoo.com>

### CODE REVIEW

**Comment by mikofski** [comment]
Hi @chetan201,

this PR closes #36 an issue that happens sometimes when a bypass diode is on, or there are some modules that are heavily shaded, or whatever, the point is that the max voltage in the IV curve for the shaded/bypasses cell had been determined by an estimate of Voc with Rs=0 and Rsh&rarr;&infin;, but if the cell is too shaded then this value is too low to be Vmax, and to match the voltages of other cells, we had to extrapolate. This caused errors, because the slope at that estimate was totally wrong.

This PR fixes that by using a constant, Vmax, estimated by Voc at STC for Vmax always, this value is guaranteed to be in the 4th quadrant always. Therefore there are never issues with extrapolating.

vocest-proof

**Comment by mikofski** [comment]
Hi @chetan201 any update on this? If you're looking for reviewers, I think @bmeyers was the one who originally reported this issue. Let me know if you have any questions. Thanks! -Mark

**Comment by chetan201** [comment]
Hi @mikofski,
I went through [https://github.com/SunPower/PVMismatch/issues/36](url) briefly and I couldn't intuitively understand what the issue is. 

If you have a short script that I can quickly use to reproduce the bug, that will be great to get me up to speed quick. Alternatively, can you suggest a shade (bypass) scenario I can implement quickly to test this?

**Comment by mikofski** [comment]
So the issue occurs if the max voltage on a parallel string isn't big enough to go around the knee in the IV curve. Then the points that get extrapolated to match the voltages of the _other_ parallel strings is wrong, because it's based on the last two points. This custom extrapolation is defined in [`pvconstants.npinterpx`](../blob/master/pvmismatch/pvmismatch_lib/pvconstants.py#L24)

For example, if I create a 10x10 PV system of SPR-E321 96-cell modules and set every cell in string no. 1 to 10[W/m^2]

```python
from pvmismatch import *
from matplotlib import pyplot as plt
plt.ion()
pvsys = pvsystem.PVsystem()
pvsys.setSuns({1: {pvmod: 0.01 for pvmod in range(10)}})
pvsys.pvstrs[1].plotStr()
plt.tight_layout()
pvsys.plotSys()
plt.tight_layout()
pvsys.Pmp
```

Then Pmp degrades from 32117.116666835867[W] to 28848.63489446646[W], due to the shading, and the shaded string's IV curve looks like this:

str1_e10

... while an unshaded string looks like this:

pvsys_1sun

Note how the IV curve turns straight down for the unshaded string, but is still in the "knee" for the shaded string. This is an artifact of the selection of voltage points chosen for that string. The max voltage is fixed at `Vmax = n * Vt * log(Igen/Isat + 1.0)` as given by the estimate of [`Voc`](../blob/master/pvmismatch/pvmismatch_lib/pvcell.py#L166-L175) at Tcell and Ee.

**Comment by mikofski** [comment]
@chetan201 does my example explain this PR and #36 better? Do you see how the linear extrapolation is just wrong if you don't select a high enough Vmax, because the slope is still changing, and you haven't actually gotten to a linear region yet? It's okay if they're wrong in a region we don't care about, but it is very likely that inverters **will** operate between Vmp and Voc, so this area is important to get correct currents. Let me know if you have more questions. Thanks!

File: pvmismatch/docs/conf.py [hunk 1]
```diff
@@ -29,7 +29,9 @@
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
 extensions = ['sphinx.ext.autodoc', 'sphinx.ext.intersphinx',
               'sphinx.ext.mathjax', 'sphinx.ext.napoleon',
-              'sphinx.ext.graphviz']
+              'sphinx.ext.graphviz', 'sphinx.ext.imgconverter']
+
+image_converter = 'c://Program Files//ImageMagick-7.0.7-Q16//convert.exe'
```
Comment (chetan201) [inline pvmismatch/docs/conf.py]: where is this program running?

Comment (mikofski) [reply pvmismatch/docs/conf.py]: [ImageMagick](https://www.imagemagick.org/script/index.php) is used by Sphinx to convert the Travis `.svg` badge into an image that can be used in a pdf and latex. I installed it, and it works. It's available on all platforms.

**Comment by mikofski** [comment]
`newton` is specifically for scalar, univariate functions, and it's pure Python, whereas `fsolve` is for multivariate, vector functions, and it's FORTRAN from MINPACK, so `fsolve` just seemed like overkill.

**Pull request merged**